kernel/src/snapshot/builder.rs

```diff
@@ -23,16 +26,34 @@ use url::Url;
 /// # Ok(())
 /// # }
 /// ```
+//
+// Note the SnapshotBuilder must have either a table_root or an existing_snapshot (but not both).
+// We enforce this in the constructors. We could improve this in the future with different
+// types/add type state.
+#[derive(Debug)]
 pub struct SnapshotBuilder {
-    table_root: Url,
+    table_root: Option<Url>,
+    existing_snapshot: Option<SnapshotRef>,
     version: Option<Version>,
+    log_tail: Vec<LogPath>,
 }
 
 impl SnapshotBuilder {
     pub(crate) fn new(table_root: Url) -> Self {
```

Comment:
maybe a follow-up task to rename these to be more in line with what calls them. So this becomes `new_for`?

Reply:
went ahead and just knocked it out: https://github.com/delta-io/delta-kernel-rs/pull/1306